test/test_utils/__init__.py

```diff
@@ -1510,6 +1510,35 @@ def get_python_version_from_image_uri(image_uri):
     python_version = python_version_search.group()
     return "py36" if python_version == "py3" else python_version
 
+def construct_buildspec_path(dlc_path, framework_path, buildspec, framework_version):
+    """
+    Construct a relative path to the buildspec yaml file by iterative checking on the existence of
+    a specific version file for the framework being tested. Possible options include:
+    [buildspec-[Major]-[Minor]-[Patch].yml, buildspec-[Major]-[Minor].yml, buildspec-[Major].yml, buildspec.yml]
+    :param dlc_path: path to the DLC test folder
+    :param framework_path: Framework folder name
+    :param buildspec: buildspec file name
+    :param framework_version: default (long) framework version name
+    """
+    if framework_version:
+        # pattern matches for example 0.3.2 or 22.3
+        pattern = r"^(\d+)?(\.\d+)?(\.\d+)$"
+        matched = re.search(pattern, framework_version)
+        if matched:
+            version = ""
+            versions = []
+            for match in matched.groups():
+                if match:
+                    version = f'{version}{match.replace(".","-")}'
+                    versions.append(version)
+
+            # skip search with only major version number
```

Comment:
Nit: Given the comment in the resolved conversation below, how is the only-major-version number skipped? Is this still a relevant comment?

Reply:
The major version is not skipped. We expect the framework version to be composed of major, minor and patch. According to @saimidu, we start with a narrower version number to a more broader version number. For instance, in the case of 1.2.3, we should search for the buildspec file in the following order: 
 buildspec-1-2-3.yml -> buildspec-1-2.yml -> buildspec-1.yml -> buildspec.yml (default buildspec file).

Reply:
The comment is misplaced. I am removing because it no longer applies in the current implementation.